### ISSUE TITLE
Code inconsistent with the paper

### ISSUE THREAD

**Issue by Wintoplay** [facebookresearch/coconut#5]
Why do you use epoch = 25 instead of 6 and 3 for GSM8K?
also the batch size don't match.

**Comment by Ber666** [comment]
I don't see any inconsistencies with the paper. 25 is the maximum epochs and 32 is the batch size per GPU. Please read argument description in the readme and the paper for more details. Let me know if there are still any questions.

**Comment by Wintoplay** [comment]
Thank for your reply
I see 32x4 global batch size.
However, for the num_epochs stated in the paper. It said that cot stage was trained for 6 epochs (GSM8k), but in args/gsm_cot.yaml use num_epochs:25

I am quite confused about that. Should I use 6 or 25?

**Comment by Ber666** [comment]
25 is the **maximum** epoch number for the training. As mentioned in the readme, you should select the checkpoint based on the validation set accuracy. In the experiment presented in our paper, the checkpoint trained for 6 epochs is selected.
